pkg/proxy/ipvs/proxier.go

```diff
@@ -629,9 +629,14 @@ func CanUseIPVSProxier() (bool, error) {
 	loadModules := sets.NewString()
 	wantModules.Insert(ipvsModules...)
 	loadModules.Insert(mods...)
-	modules := wantModules.Difference(loadModules).List()
-	if len(modules) != 0 {
-		return false, fmt.Errorf("Failed to load kernel modules: %v", modules)
+	modules := wantModules.Difference(loadModules)
+
+	// Now check if the missing kernel modules are there to be loaded
+	for kmod := range modules {
+		err := utilexec.New().Command("modprobe", "--dry-run", kmod).Run()
```

Comment:
Please check my test:

```
# cat /proc/modules | grep ip_vs
ip_vs_dh 12688 0 - Live 0xffffffffa05e5000
ip_vs_lc 12516 0 - Live 0xffffffffa05d5000
ip_vs_sh 12688 0 - Live 0xffffffffa05c5000
ip_vs_wrr 12697 0 - Live 0xffffffffa05c0000
ip_vs_rr 12600 4 - Live 0xffffffffa05bb000
ip_vs 141092 14 ip_vs_dh,ip_vs_lc,ip_vs_sh,ip_vs_wrr,ip_vs_rr, Live 0xffffffffa0592000
nf_conntrack 111302 7 nf_conntrack_netlink,ip_vs,nf_nat_masquerade_ipv4,nf_nat_ipv4,nf_nat,nf_conntrack_ipv4,xt_conntrack, Live 0xffffffffa045a000
libcrc32c 12644 2 ip_vs,dm_persistent_data, Live 0xffffffffa0349000

# modprobe --dry-run ip_vs
modprobe: ERROR: ../libkmod/libkmod.c:557 kmod_search_moddep() could not open moddep file '/lib/modules/3.10.0-514.26.2.el7.x86_64/modules.dep.bin'
```

The updated logic doesn't hit my expectation.

Reply:
I would suggest:

Firstly, modprobe ip_vs_* modules, then just log warnings if error occured.

Secondly,  check fields in `/proc/modules`, then return error if required filed missing.

Reply:
Have you mounted /lib/modules from the node to container?
`check fields in /proc/modules, then return error if required filed missing`
This approach makes sense to me.

Reply:
> Have you mounted /lib/modules from the node to container?

NO.